indexing-service/src/main/java/io/druid/indexing/worker/http/WorkerResource.java

```diff
@@ -54,18 +57,60 @@
   private final Worker enabledWorker;
   private final Worker disabledWorker;
   private final WorkerCuratorCoordinator curatorCoordinator;
-  private final ForkingTaskRunner taskRunner;
+  private final TaskRunner taskRunner;
 
   @Inject
   public WorkerResource(
       Worker worker,
       WorkerCuratorCoordinator curatorCoordinator,
-      ForkingTaskRunner taskRunner
+      TaskRunner taskRunner
 
   ) throws Exception
   {
     this.enabledWorker = worker;
-    this.disabledWorker = new Worker(worker.getHost(), worker.getIp(), worker.getCapacity(), DISABLED_VERSION);
+    this.disabledWorker = new Worker(
+        worker.getHost(),
+        worker.getIp(),
+        worker.getCapacity(),
+        DISABLED_VERSION,
+        enabledWorker.getLastCompletedTaskTime()
+    )
+    {
+      @Override
+      @JsonProperty
+      public String getHost()
+      {
+        return enabledWorker.getHost();
```

Comment:
Also in light of the need for overriding it'd probably be cleaner to generate the `disabledWorker` every time someone posts a disable call. That happens so infrequently that there is not really much need to optimize it…